src/runtime/internal/utils.ts

```diff
@@ -120,4 +120,8 @@ export function set_store_value(store, ret, value = ret) {
 	return ret;
 }
 
-export const has_prop = (obj, prop) => Object.prototype.hasOwnProperty.call(obj, prop);
\ No newline at end of file
+export const has_prop = (obj, prop) => Object.prototype.hasOwnProperty.call(obj, prop);
+
+export function action_destroyer(action_result) {
```

Comment:
I renamed this to `action_destroyer` (rather than `destroy_action`) because the function doesn't actually destroy the action when called. I also moved the check for whether `action_result.destroy` is a function out, so it's called right away. Do these changes make sense to you?

Reply:
oh yeah. it is simpler now :D